src/utils/vcs.rs

```diff
@@ -207,6 +207,13 @@ impl VcsUrl {
     }
 }
 
+fn extract_provider_name(host: &str) -> String {
+    let trimmed = host.trim_end_matches('.');
+    let mut iter = trimmed.rsplit('.');
+    iter.next(); // skip TLD
+    iter.next().unwrap_or(trimmed).to_owned()
```

Comment:
[nit] I believe you can do this for even more simplicity:

```suggestion
    trimmed.rsplit('.').nth(1).unwrap_or(trimmed)
```

See https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.nth